include/git2/sys/transport.h

```diff
@@ -32,6 +32,11 @@ typedef enum {
 	GIT_TRANSPORTFLAGS_NONE = 0,
 } git_transport_flags_t;
 
+typedef struct {
+	const git_remote_head * const *refs;
+	size_t count;
+} git_fetch_negotiation;
```

Comment:
This is somewhat weird because, for the first time, we're the ones passing "sys-level" structs to "userspace", expecting that they might (or might not) be able to understand some of those options. I've refrained from adding a `version` field, because it's "feature-based", hence a monotonic number might not be sufficient.